07_errors/n0npax/leveldbStore.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+var levelDBPath = "/tmp/leveldb"
+
+// NewLevelDBStorer creates new storer for leveldb
+func NewLevelDBStorer() *LevelDBStore {
+	db, err := leveldb.OpenFile(levelDBPath, nil)
+	dbErrorPanic(err)
+	return &LevelDBStore{nextID: 0, ldb: db}
+}
+
+// CreatePuppy creates puppy
+func (l *LevelDBStore) CreatePuppy(puppy *Puppy) (int, error) {
+	l.Lock()
+	defer l.Unlock()
+	id, err := l.putPuppy(l.nextID, puppy)
```

Comment:
This is not setting the ID of `puppy` to `l.nextID`, but your tests are all passing. You should see why this is and add a test case for it

Reply:
fixed. It was already fixed in lab09 and lab10.

Reply:
tests also added